src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs

```diff
@@ -0,0 +1,130 @@
+                Locations = tLevel.Locations
+                    .Select(l => (Venue: venues[l.VenueId], Location: l))
+                    .Select(t => new TLevelLocationViewModel()
+                    {
+                        TLevelLocationId = t.Location.TLevelLocationId,
+                        AddressLine1 = t.Venue.AddressLine1,
+                        AddressLine2 = t.Venue.AddressLine2,
+                        County = t.Venue.County,
+                        Email = t.Venue.Email,
+                        Latitude = Convert.ToDecimal(t.Venue.Latitude),
+                        Longitude = Convert.ToDecimal(t.Venue.Longitude),
+                        Postcode = t.Venue.Postcode,
+                        Telephone = t.Venue.Telephone,
+                        Town = t.Venue.Town,
+                        VenueName = t.Venue.VenueName,
+                        Website = EnsureHttpPrefixed(t.Venue.Website)
+                    })
+                    .ToArray(),
+                OfferingType = Core.Search.Models.FindACourseOfferingType.TLevel,
+                Provider = new ProviderViewModel()
+                {
+                    ProviderName = sqlProvider.DisplayName,
+                    Ukprn = provider.UnitedKingdomProviderReferenceNumber,
+                    AddressLine1 = providerContact?.ContactAddress?.SAON?.Description,
+                    AddressLine2 = providerContact?.ContactAddress?.PAON?.Description,
+                    Town = providerContact?.ContactAddress?.Items?.FirstOrDefault()?.ToString(),
+                    Postcode = providerContact?.ContactAddress?.PostCode,
+                    County = providerContact?.ContactAddress?.Locality,
+                    Telephone = providerContact?.ContactTelephone1,
+                    Fax = providerContact?.ContactFax,
+                    Website = EnsureHttpPrefixed(providerContact?.ContactWebsiteAddress),
+                    Email = providerContact?.ContactEmail,
+                    EmployerSatisfaction = feChoice?.EmployerSatisfaction,
+                    LearnerSatisfaction = feChoice?.LearnerSatisfaction
+                },
+                Qualification = new QualificationViewModel()
+                {
+                    FrameworkCode = tLevel.TLevelDefinition.FrameworkCode,
+                    ProgType = tLevel.TLevelDefinition.ProgType,
+                    QualificationLevel = "3",
+                    TLevelName = tLevel.TLevelDefinition.Name
+                },
+                StartDate = tLevel.StartDate,
+                StudyMode = CourseStudyMode.FullTime,
+                TLevelId = tLevel.TLevelId,
+                Website = tLevel.Website,
+                WhatYouCanDoNext = tLevel.WhatYouCanDoNext,
+                WhatYoullLearn = tLevel.WhatYoullLearn,
+                WhoFor = tLevel.WhoFor
+            };
+        }
+
+        private static string EnsureHttpPrefixed(string url) => !string.IsNullOrEmpty(url)
```

Comment:
This is duplicated from CourseDetail... what to do with it I wonder? Extension method on string, scoped to FAC? Static within FAC? (not named Helper, 😄) I'm not sure.

Reply:
I think this used to be in an `HttpUtil` class. I'm open to naming suggestions but I'm not crazy about making it an extension method on `string`. Could make it an extension method on `Flurl.Url` and use that here (we use Flurl elsewhere already)?

Reply:
Your call, happy to come back to it later if the solution isn't apparent right now.